webserver/webserver/src/main/java/io/helidon/webserver/ByteBufDataChunk.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.webserver;
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import io.helidon.common.http.DataChunk;
+
+import io.netty.buffer.ByteBuf;
+
+/**
+ * A special DataChunk implementation based on Netty's buffers. This is used by
+ * our Jersey SPI implementation to take advantage of Netty's buffer pooling.
+ */
+public class ByteBufDataChunk implements DataChunk {
+
+    private final ByteBuf[] byteBufs;
+    private final boolean flush;
+    private final boolean readOnly;
+    private final Runnable releaseCallback;
+    private boolean isReleased = false;
+    private CompletableFuture<DataChunk> writeFuture;
+
+    /**
+     * Creates an instance given an array of {@code ByteBuf}'s.
+     *
+     * @param flush a signal that this chunk should be written and flushed from any cache if possible
+     * @param readOnly marks this buffer as read only
+     * @param byteBufs the data for this chunk. Should not be reused until {@code releaseCallback} is used
+     */
+    public ByteBufDataChunk(boolean flush, boolean readOnly, ByteBuf... byteBufs) {
```

Comment:
This does not comply with the dev [guidelines](https://github.com/oracle/helidon/blob/master/DEV-GUIDELINES.md#builders), (no public constructors on public classes)

Reply:
Do you think it would be acceptable to just have a static `create` method like `DataChunk`?

Reply:
Yes.